sdk/src/types/block/output/unlock_condition/mod.rs

```diff
@@ -24,22 +25,51 @@ pub use self::{
     storage_deposit_return::StorageDepositReturnUnlockCondition, timelock::TimelockUnlockCondition,
 };
 use crate::types::block::{
-    address::{Address, AddressCapabilityFlag, RestrictedAddress},
+    address::{Address, AddressCapabilityFlag, AddressError, RestrictedAddress},
     output::{
         feature::NativeTokenFeature, AccountOutput, AnchorOutput, DelegationOutput, NftOutput, StorageScore,
         StorageScoreParameters,
     },
     protocol::{CommittableAgeRange, ProtocolParameters, WorkScore},
     slot::SlotIndex,
-    Error,
 };
 
+#[derive(Debug, PartialEq, Eq, derive_more::Display, derive_more::From)]
+#[allow(missing_docs)]
+pub enum UnlockConditionError {
+    #[display(fmt = "invalid unlock condition kind: {_0}")]
+    InvalidUnlockConditionKind(u8),
+    #[display(fmt = "invalid unlock condition count: {_0}")]
+    InvalidUnlockConditionCount(<UnlockConditionCount as TryFrom<usize>>::Error),
+    #[display(fmt = "expiration unlock condition with milestone index and timestamp set to 0")]
+    ExpirationUnlockConditionZero,
+    #[display(fmt = "timelock unlock condition with milestone index and timestamp set to 0")]
+    TimelockUnlockConditionZero,
+    #[display(fmt = "unlock conditions are not unique and/or sorted")]
+    UnlockConditionsNotUniqueSorted,
+    #[display(fmt = "disallowed unlock condition at index {index} with kind {kind}")]
+    DisallowedUnlockCondition { index: usize, kind: u8 },
+    #[display(fmt = "missing slot index")]
```

Comment:
So this one can be removed?

Reply:
How is it different?

Reply:
it would be "UnlockConditionsNotUniqueSorted"

Reply:
Gh selection being annoying, my comment was about ` #[display(fmt = "missing slot index")]`